pkg/storaged/utils.js

```diff
@@ -172,15 +172,15 @@ export function mdraid_name(mdraid) {
     if (!mdraid.Name)
         return "";
 
-    var parts = mdraid.Name.split(":");
+    const parts = mdraid.Name.split(":");
 
     if (parts.length != 2)
         return mdraid.Name;
 
     /* if we call hostnamed too early, before the dbus.proxy() promise is fulfilled,
      * it will not be valid yet; it's too inconvenient to make this
      * function asynchronous, so just don't show the host name in this case */
-    if (hostnamed.StaticHostname === undefined || parts[0] == hostnamed.StaticHostname)
+    if (hostnamed.StaticHostname === undefined || parts[0] == hostnamed.StaticHostname || parts[0] == hostnamed.Hostname)
```

Comment:
There's an awful lot going on in that line now, and some of the things are of a different nature.

Is it possible that `hostnamed.Hostname` will also be undefined?

At the very least, this comment should probably be expanded from explaining what the first item is attempting to do to explaining what all 3 items are attempting to do, and what the expected result (ie: showing only the name) is.

Reply:
Yes, it's possible that hostnamed.Hostname is undefined, during the time when hostname proxy is not initialized yet (as the comment says). In that case it wants to treat all detected RAIDs as local. That behaviour does not change, it just changes the "proxy is initialized" case. If Hostname is undefined, the new comparison is always false, and thus it's a no-op.

Reply:
I.e. do you want me to add `hostnamed.Hostname === undefined`? I'm not sure if that's a good idea, a machine might legitimately not have a transient host name and never get one, this might break the actual name detection.

@allisonkarlitskaya, do you have a suggestion how to rewrite this? I'm honestly not sure.

Reply:
I think I'm mostly looking for a bigger comment, which explains:

 - what this does (ie: volume name without hostname, mostly for the "on this machine" case)
 - workaround for not having the hostname yet (`undefined`)
 - the other two cases being the static and transient name, and what that means.  i had no idea this was even possible.

Reply:
Makes sense. How about this?

    /* if we call hostnamed too early, before the dbus.proxy() promise is
     * fulfilled, it will not be valid yet (hostnamed properties are undefined);
     * it's too inconvenient to make this function asynchronous, so just don't
     * show the host name in this case. Otherwise we check the static (from
     * /etc/hostname) and transient (acquired from DHCP server via
     * NetworkManager → hostnamed, may not exist) host name -- if either one matches, we
     * consider the RAID a local one and just show the device name.
     * Otherwise it's a remote one, and include the host in the name */

Reply:
Version 2:

    /* Check the static (from /etc/hostname) and transient (acquired from DHCP server via
     * NetworkManager → hostnamed, may not exist) host name -- if either one matches, we
     * consider the RAID a local one and just show the device name.
     * Otherwise it's a remote one, and include the host in the name.
     *
     * However: if we call hostnamed too early, before the dbus.proxy() promise is
     * fulfilled, it will not be valid yet (hostnamed properties are undefined);
     * it's too inconvenient to make this function asynchronous, so just don't
     * show the host name in this case. */

Reply:
much nicer, thanks